app.py

```diff
@@ -0,0 +1,158 @@
+            file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
+
+            # get the uid from the file name
+            uid_for_user = filename.split('_')[2].split('.')[0]
+
+            return render_template('index.html', message='File uploaded successfully. ', uid=uid_for_user)
+
+    return render_template('index.html')
+
+
+def allowed_file(filename):
+    # Specify the allowed file extensions
+    allowed_extensions = {'pptx'}
+
+    # Check if the file extension is allowed
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in allowed_extensions
+
+
+def generate_unique_filename(filename):
+    # Generate a unique filename by combining the original filename, timestamp, and UID
+    timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
+    uid = str(uuid.uuid4().hex)  # Generate a random UID
+    filename_without_extension, extension = os.path.splitext(filename)
+    new_filename = f"{filename_without_extension}_{timestamp}_{uid}{extension}"
+    return new_filename
+
+
+def get_file_status(filename):
+    # Check if the file exists in outputs folder, pending, else return not found
+    pptx_filename = os.path.splitext(filename)[0] + '.pptx'
+    json_filename = os.path.splitext(filename)[0] + '.json'
+    if os.path.exists(os.path.join(DONE_FOLDER, pptx_filename)):
+        return 'done'
+    elif os.path.exists(os.path.join(PENDING_FOLDER, pptx_filename)) \
+            or os.path.exists(os.path.join(UPLOAD_FOLDER, pptx_filename)):
+        return 'pending'
+    elif os.path.exists(os.path.join(DONE_FOLDER, json_filename)):
+        return 'done'
+    else:
+        return 'not found'
+
+
+def get_file_details(filename):
+    filename_without_extension, _ = os.path.splitext(filename)
+    timestamp, uid = filename_without_extension.rsplit('_', 2)[-2:]
+
+    original_filename = filename_without_extension.split('_', 1)[0]
+    date_and_time = format_timestamp(timestamp)
+    return {
+        'status': get_file_status(filename),
+        'filename': original_filename,
+        'timestamp': date_and_time,
+        'explanation': get_file_explanation(filename)
+    }
+
+
+def get_file_explanation(filename):
+    # Retrieve the explanation from the processed output file if available
+    output_file_path = os.path.join(DONE_FOLDER, os.path.splitext(filename)[0] + '.json')
+    if os.path.exists(output_file_path):
+        # Read the data from the JSON file
+        with open(output_file_path, 'r') as file:
+            json_data = file.read()
+            data = json.loads(json_data)
+            return data
+
+    return None
+
+
+class UIDNotFoundException(Exception):
+    pass
+
+
+def find_file_by_uid(uid):
+    folders = ['uploads', 'pending', 'outputs']
+    for folder in folders:
+        for root, dirs, files in os.walk(folder):
+            for file in files:
+                if uid in file:
+                    return os.path.basename(file)
+    raise UIDNotFoundException("UID not found")

```

Comment:
Your loops have other reasons to fail.
raising an exception without being sure of it is not good.